qubesadmin/tools/qvm_start_daemon.py

```diff
@@ -785,19 +785,20 @@ def main(args=None):
 
             events_listener = asyncio.ensure_future(events.listen_for_events())
 
-            for signame in ('SIGINT', 'SIGTERM'):
-                loop.add_signal_handler(getattr(signal, signame),
-                                        events_listener.cancel)  # pylint: disable=no-member
-
-            loop.add_signal_handler(signal.SIGHUP,
-                                    launcher.send_monitor_layout_all)
-
-            conn = xcffib.connect()
-            x_watcher = XWatcher(conn, args.app)
-            x_fd = conn.get_file_descriptor()
-            loop.add_reader(x_fd, x_watcher.event_reader,
-                            events_listener.cancel)
-            x_watcher.update_keyboard_layout()
+            if "guivm" in enabled_services:
+                for signame in ('SIGINT', 'SIGTERM'):
+                    loop.add_signal_handler(getattr(signal, signame),
+                                            events_listener.cancel)  # pylint: disable=no-member
```

Comment:
This should be done whenever events_listener is started, not only in guivm.

Reply:
Thanks, should be ok with the new commit